### PR TITLE
Allows PSM2 gets its TransportKey from PMIx

### PR DESCRIPTION
**Pull request by marisaroman-cornelisnetworks**
Author: Ralph Castain <rhc@pmix.org>

Signed-off-by: Marisa Roman <marisa.roman@cornelisnetworks.com>

### CODE REVIEW

**Comment by marisaroman-cornelisnetworks** [comment]
@mwheinz : I added a pull request draft. If I have an ok from Ralph I will open a formal pull request.

File: ompi/mca/mtl/psm2/mtl_psm2.c [hunk 1]
```diff
@@ -101,7 +103,17 @@ int ompi_mtl_psm2_module_init(int local_rank, int num_local_procs) {
...
+    opal_process_name_t pname;
+
+    generated_key = NULL;
+    pname.jobid = opal_process_info.my_name.jobid;
+    pname.vpid = OPAL_VPID_WILDCARD;
+    OPAL_MODEX_RECV_VALUE_OPTIONAL(rc, PMIX_CREDENTIAL, &pname,
+                                    (char**)&generated_key, PMIX_STRING);
+
+    if (PMIX_SUCCESS != rc || NULL == generated_key) {
+        generated_key = getenv("OMPI_MCA_orte_precondition_transports");
```
Comment (rhc54) [inline ompi/mca/mtl/psm2/mtl_psm2.c:115]: You need to "strdup" the `getenv` value here (checking for NULL first) so you can `free` it later

**Comment by rhc54** [comment]
You just need to fix the problem I cited above and then this is fine

File: ompi/mca/mtl/psm2/mtl_psm2.c (hunk 1 above)
Comment (marisaroman-cornelisnetworks) [reply ompi/mca/mtl/psm2/mtl_psm2.c:115]: Thanks

**Comment by jsquyres** [comment]
@marisaroman-cornelisnetworks FYI -- you didn't need to close this PR.  Just push updated commits to your fork's branch, and github PR's will auto-update to show the current state of that branch.  Also note that you can convert from Draft to regular PR once it's ready (and you can also convert back to Draft if needed).

**Comment by marisaroman-cornelisnetworks** [comment]
Thanks! you are right! I can reopen correct?

**Comment by jsquyres** [comment]
> Thanks! you are right! I can reopen correct?

Yes

**Pull request closed without merging**